### ISSUE TITLE
[css-transitions] undesired behaviour - transition of inherited value happening in sequence.

### ISSUE THREAD

**Issue by meirkoen** [w3c/csswg-drafts#5324]
Transition of inherited value, such as `text-indent`, in nested elements, is happening in sequence and not simultaneously with the transition of the parent value. 

Here is a codepen demonstrates the issue: [https://codepen.io/meirkoen/pen/mdVGRYj?editors=1100](https://codepen.io/meirkoen/pen/mdVGRYj?editors=1100)
On the left side you can see 3 variants of the issue. 
On the right side you can see how it can be fixed - by specifying the values on the children as well.

I have tried to search the [[css-transitions]](https://drafts.csswg.org/css-transitions/) spec for a definition of this behaviour but I did not see one.

My guess is that this is an undesired behaviour (bug) resulting by this section in the spec [https://drafts.csswg.org/css-transitions/#starting](https://drafts.csswg.org/css-transitions/#starting) that has left unhandled.

Before I will propose anything I wanted to post the issue to be sure that I'm not mistaken.

**Comment by Loirooriol** [comment]
Seems a Chromium/WebKit bug to me. Firefox does it well.

**Comment by meirkoen** [comment]
> Seems a Chromium/WebKit bug to me. Firefox does it well.

I have tested it on Chrome and Safari and it’s the same so I guess it's WebKit.

However Firefox doesn’t handle the `background-color` transition well (the deepest span element become darker first).

I think there is an ambiguity in the [css-transitions] spec regarding transition of inherited value, which caused this bug in the implementation.

Could we add to the spec clear definition of what should happen?

I will file a bug in WebKit as well.

**Comment by Loirooriol** [comment]
> However Firefox doesn’t handle the `background-color` transition well

This seems because mid-way the transition you are stacking `rgba(255, 255, 0, 0.5)` on top of `rgb(128, 128, 0)`, so it looks more yellow. I don't know enough about colors to tell if that's the correct way to interpolate with `transparent`, or if the spans should actually be `rgba(128, 128, 0, 0.5)`.

**Comment by meirkoen** [comment]
> > However Firefox doesn’t handle the `background-color` transition well
> 
> This seems because mid-way the transition you are stacking `rgba(255, 255, 0, 0.5)` on top of `rgb(128, 128, 0)`, so it looks more yellow.

Thanks, I've added initial value `background-color: inherit;` in the [codpen](https://codepen.io/meirkoen/pen/mdVGRYj?editors=1100), to show the bug in properties with default value other than `inherit`

**Comment by meirkoen** [comment]
Link to the WebKit bug [https://bugs.webkit.org/show_bug.cgi?id=214352](https://bugs.webkit.org/show_bug.cgi?id=214352)

Closing the issue.

**Comment by kizu** [comment]
I stumbled upon this problem recently, and when investigating it, I found this closed issue.

There seem to be older issues in the browser bug trackers regarding this (https://bugs.webkit.org/show_bug.cgi?id=46041 and https://issues.chromium.org/issues/40337057), and while the spec explicitly allows browsers to do whatever they want there, I think this is something we'd want to revisit: after all, this place in the specs is there from back in CSS3.

I also wrote a blog post with a few more details: https://blog.kizu.dev/transitions-of-inherited-properties/

**Comment by bramus** [comment]
I rely on this exact delayed behavior for https://www.bram.us/2023/10/23/css-scroll-detection/: because the properties inherit with a delay, I’m able to determine which direction the user is scrolling into.

(This hack will eventually become obsolete with the availability of Scroll State Queries)

**Comment by kizu** [comment]
Yes, I think there are use cases for both behaviors. The delayed inheritance allows for some cool effects, but in the UI often you want it to be instant.

What if (names to be bikeshed): `transition-inheritance: instant | delayed`? Not sure about the default value, but, from what I saw, people consider the delayed one to be a bug more often. We could also have `auto`, where browsers are free to continue doing what they're doing now. But it would be much better to have a more stable default value.
